gradle-idea-configuration/src/main/groovy/com/palantir/gradle/ideaconfiguration/IdeaConfigurationExtension.java

```diff
@@ -17,31 +17,7 @@
 package com.palantir.gradle.ideaconfiguration;
 
 import org.gradle.api.NamedDomainObjectContainer;
-import org.gradle.api.Project;
 
-public class IdeaConfigurationExtension {
-    private final NamedDomainObjectContainer<PluginDependency> externalDependencies;
-
-    public IdeaConfigurationExtension(Project project) {
-        this.externalDependencies = project.container(PluginDependency.class, PluginDependency::new);
```

Comment:
Using an abstract class and abstract property, Gradle will just construct the type for us (see https://github.com/palantir/gradle-guide/blob/develop/guide/managed-types-and-properties.md)

Reply:
TODO: Write an errorprone to check that people don't do this and just make abstract types instead.